test/integration/load/load_test.go

```diff
@@ -171,9 +178,94 @@ func TestScaleDeployment(t *testing.T) {
 	if testConfig.Cleanup {
 		err = kubernetes.MustDeleteDeployment(ctx, deploymentsClient, deployment)
 		require.NoError(t, err, "error deleteing load deployment")
+		err = kubernetes.WaitForPodsDelete(ctx, clientset, namespace, podLabelSelector)
+		require.NoError(t, err, "error waiting for pods to delete")
 	}
 }
 
+func TestValidCNSStateDuringScaleAndCNSRestart(t *testing.T) {
```

Comment:
This looks like the node restart scenario. We need to ensure that we have pods on the cluster, validate, restart CNS, validate. No need to scale down or use `skipWait` on scale up.

Or is this adding additional coverage?

Reply:
This adds additional coverage, by restarting CNS in the middle of a scale down event
There was a bug before in CNS, that on restart of CNS, and scaling up/down, IPs could get "leaked", i.e., their Pods are gone, but the IPs in CNS are still "allocated", so now we're testing that here

Reply:
I'll add in a "ToTriggerDropgzRestart" to function name and add comments to describe why we restart cns during a pod scaling even (to trigger dropgz)